libsyclinterface/source/dpctl_sycl_platform_interface.cpp

```diff
@@ -269,3 +270,46 @@ size_t DPCTLPlatform_Hash(__dpctl_keep const DPCTLSyclPlatformRef PRef)
         return 0;
     }
 }
+
+__dpctl_give DPCTLDeviceVectorRef
+DPCTLPlatform_GetDevices(__dpctl_keep const DPCTLSyclPlatformRef PRef,
+                         DPCTLSyclDeviceType DTy)
+{
+    auto P = unwrap<platform>(PRef);
+    if (!P) {
+        error_handler("Cannot retrieve devices from DPCTLSyclPlatformRef as "
+                      "input is a nullptr.",
+                      __FILE__, __func__, __LINE__);
+        return nullptr;
+    }
+
+    using vecTy = std::vector<DPCTLSyclDeviceRef>;
+    vecTy *DevicesVectorPtr = nullptr;
+    try {
+        DevicesVectorPtr = new vecTy();
+    } catch (std::exception const &e) {
+        delete DevicesVectorPtr;
+        error_handler(e, __FILE__, __func__, __LINE__);
+        return nullptr;
+    }
+
+    // handle unknown device
```

Comment:
I believe it might be helpful to place a comment somewhere (and/or cover in tests) that `automatic` and `custom` device type are handling as `unknown`.

Reply:
That's a good idea

Reply:
I've added a comment mentioning this